core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -113,4 +113,21 @@ public static List<String> manifestListLocations(Table table) {
     }
     return manifestListLocations;
   }
+
+  /**
+   * Returns locations of statistics files in a table.
+   *
+   * @param table table for which statistics files needs to be listed
+   * @return the location of statistics files
+   */
+  public static List<String> statisticsFilesLocations(Table table) {
+    List<String> statisticsFilesLocations = Lists.newArrayList();
+    TableOperations ops = ((HasTableOperations) table).operations();
```

Comment:
Looks like this depends on the public API. Let's get that one in first so we don't need to go around the API here.

Reply:
@rdblue here I am following existing code in the `ReachableFileUtil` class 

https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/core/src/main/java/org/apache/iceberg/ReachableFileUtil.java#L62-L64

since this class already casts to `HasTableOperations`, I assumed it's OK to do that and therefore removed public API changes from this PR.
If you want first merge Table API changes, the https://github.com/apache/iceberg/pull/4741 is ready for your review